app/src/main/java/com/github/featuredetectandroid/MainActivity.kt

```diff
@@ -1,27 +1,146 @@
 package com.github.featuredetectandroid
 
+import android.Manifest
+import android.content.pm.PackageManager
 import android.os.Bundle
+import android.util.Log
+import android.util.Size
+import android.widget.Toast
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.activity.viewModels
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageAnalysis.OUTPUT_IMAGE_FORMAT_YUV_420_888
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.compose.foundation.Image
 import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.material.FloatingActionButton
+import androidx.compose.material.Text
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import com.github.featuredetectandroid.ui.theme.Theme
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.core.content.ContextCompat
+import com.github.featuredetectandroid.ui.GrayscaleViewModel
+import com.github.featuredetectandroid.ui.theme.FeatureDetectAppTheme
+import com.github.featuredetectandroid.utils.PhotoAnalyzer
+import java.util.concurrent.Executors
+
+private const val TAG = "MainActivity"
+
+private const val RESOLUTION_WIDTH = 640
+private const val RESOLUTION_HEIGHT = 360
 
 class MainActivity : ComponentActivity() {
+    private val cameraExecutor = Executors.newSingleThreadExecutor()
+    private val imageViewModel by viewModels<GrayscaleViewModel>()
+    private lateinit var cameraPermission: ActivityResultLauncher<String>
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        cameraPermission =
+            registerForActivityResult(ActivityResultContracts.RequestPermission()) { granted ->
+                if (granted) {
+                    Toast.makeText(
+                        this,
+                        "Permissions granted.",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                    Log.d(TAG, "Permission was granted successfully.")
+                    startCamera()
+                } else {
+                    Toast.makeText(
+                        this,
+                        "Permissions not granted by the user.",
+                        Toast.LENGTH_SHORT
+                    ).show()
```

Comment:
You can replace "Permissions not granted by the user." with "Permissions not granted." for uniformity and also squeeze it into a single line

Reply:
fixed